runtime/integration-tests/src/cases/liquidity_pools_gateway_queue.rs

```diff
@@ -1,41 +1,128 @@
+use runtime_common::account_conversion::AccountConverter;
 use sp_runtime::traits::One;
 
 use crate::{
+	cases::liquidity_pools::{
+		utils::*, AUSD_CURRENCY_ID, DEFAULT_BALANCE_GLMR, DEFAULT_DOMAIN_ADDRESS_MOONBEAM,
+		DOMAIN_MOONBEAM, GLMR_CURRENCY_ID, POOL_ID,
+	},
 	config::Runtime,
 	env::{Blocks, Env},
 	envs::fudge_env::{FudgeEnv, FudgeSupport},
-	utils::{currency::cfg, genesis, genesis::Genesis},
+	utils::{accounts::Keyring, currency::cfg, genesis, genesis::Genesis},
 };
 
-#[test_runtimes(all)]
-fn submit_and_process<T: Runtime + FudgeSupport>() {
-	let mut env = FudgeEnv::<T>::from_parachain_storage(
-		Genesis::default()
-			.add(genesis::balances::<T>(cfg(1_000)))
-			.storage(),
-	);
-
-	let expected_event = env.parachain_state_mut(|| {
-		let message = pallet_liquidity_pools::Message::AddPool { pool_id: 1 };
-		let nonce = <T as pallet_liquidity_pools_gateway_queue::Config>::MessageNonce::one();
-
-		assert_ok!(
-			<pallet_liquidity_pools_gateway_queue::Pallet<T> as MessageQueueT>::submit(
-				message.clone()
-			)
+mod inbound {
+	use super::*;
+
+	/// This test is basically `increase_deposit_request` but instead of
+	/// handling the message directly via the LP pallet, we submit it via
+	/// the LP Gateway Queue and confirm that it's processed accordingly.
+	#[test_runtimes([development])]
+	fn success<T: Runtime + FudgeSupport>() {
+		let mut env = FudgeEnv::<T>::from_parachain_storage(
+			Genesis::default()
+				.add(genesis::balances::<T>(cfg(1_000)))
+				.add(genesis::tokens::<T>(vec![(
+					GLMR_CURRENCY_ID,
+					DEFAULT_BALANCE_GLMR,
+				)]))
+				.storage(),
 		);
 
-		let stored_message = pallet_liquidity_pools_gateway_queue::MessageQueue::<T>::get(nonce);
+		setup_test(&mut env);
+
+		let expected_event = env.parachain_state_mut(|| {
+			let pool_id = POOL_ID;
+			let amount = 10 * decimals(12);
+			let investor =
+				AccountConverter::domain_account_to_account(DOMAIN_MOONBEAM, Keyring::Bob.id());
+			let currency_id = AUSD_CURRENCY_ID;
+			let currency_decimals = currency_decimals::AUSD;
+
+			// Create new pool
+			create_currency_pool::<T>(pool_id, currency_id, currency_decimals.into());
+
+			// Set permissions and execute initial investment
+			do_initial_increase_investment::<T>(pool_id, amount, investor.clone(), currency_id);
+
+			// Verify the order was updated to the amount
+			assert_eq!(
+				pallet_investments::Pallet::<T>::acc_active_invest_order(
+					default_investment_id::<T>(),
+				)
+				.amount,
+				amount
+			);
+
+			// Increasing again should just bump invest_amount
+			let nonce = <T as pallet_liquidity_pools_gateway_queue::Config>::MessageNonce::one();
+			let message = GatewayMessage::Inbound {
+				domain_address: DEFAULT_DOMAIN_ADDRESS_MOONBEAM,
+				message: Message::DepositRequest {
+					pool_id,
+					tranche_id: default_tranche_id::<T>(pool_id),
+					investor: investor.clone().into(),
+					currency: general_currency_index::<T>(currency_id),
+					amount,
+				},
+			};
```

Comment:
- Could we use a simple `Message::Invalid` and avoid all this setup?

Reply:
That will trigger a failure but it will mean that the message did reach the LP pallet. I can drop the success/failure tests and just have 2 tests one for inbound and one for outbound, and we confirm that the message is processed one way or another by LP / LP gateway.

Reply:
I'm curious, why does it trigger a failure?

Reply:
Because we'll be reaching [this](https://github.com/centrifuge/centrifuge-chain/blob/916c1e3a5182fb7194d1d9e6bc8c4e4f1ff8e42c/pallets/liquidity-pools/src/lib.rs#L1032).

Reply:
Ah ok, in the case of an inbound message, yeah. But as you said that's the expected behavior, so the tests have passed 👍🏻

Reply:
Yes, I guess that'll do. We are testing to confirm that the message reaches the destination. Will add something similar for outbound and I guess that should suffice.

Reply:
Done.